### PR TITLE
FP-349: Main menu - frontend.

### PR DESCRIPTION
**Pull request by robherba**
# Ticket(s)

- [FP-349: Main menu - frontend](https://fourkitchens.clickup.com/t/36718269/FP-349)

## Purpose

**This PR does the following:**
Adds a main menu component.

### Functional Testing

- [x] Run `npm install && npm run dev`
- [x] Go to main menu page.
- [ ] Validates that the component looks and works as expected

### Accessibility

- [ ] Navigate the secondary menu using a screen reader.
- [ ] Verify the functionality when expanding/collapsing a link with a submenu.

### Notes

- [Main menu design](https://www.figma.com/file/HwazkcYwlIm9Yz6NwIQ1b3/Web-Standards?node-id=1562%3A1151&mode=dev)

https://github.com/fourkitchens/forcepoint-shared-components/assets/9342274/2c53769e-c3cc-4745-b69e-d9e0a519bc98

### CODE REVIEW

**Comment by fkbender** [comment]
Task linked: [FP-349 Header main navigation - frontend](https://app.clickup.com/t/36718269/FP-349)

File: src/lib/components/02-components/menus/main-menu/main-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+    item: MenuItemProps,
+    index: number,
+  ): void => {
+    if (item.below && window.innerWidth >= 1156) {
+      setActive(index);
+      const selectedRect = event.currentTarget?.getBoundingClientRect();
+      const navHighlight = document.querySelector(
+        '.nav-highlight',
+      ) as HTMLElement;
+      console.log('selectedRect', selectedRect)
```
Comment (mcortes19) [inline src/lib/components/02-components/menus/main-menu/main-menu.tsx:192]: Remove this console.log

File: src/lib/components/02-components/menus/main-menu/main-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+  ctaButton?: string;
+  description?: string;
+  display: MenuItemDisplay;
+  icon?: MenuItemImageProps;
+  iconOnHover?: MenuItemImageProps;
+  linkProps?: LinkProps;
+  title: string;
+  url?: string;
+  width?: MenuItemWidth;
+  size?: any;
```
Comment (mcortes19) [inline src/lib/components/02-components/menus/main-menu/main-menu.tsx:62]: Does this prop need to be `any`

File: src/lib/components/02-components/menus/main-menu/main-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+    parent: 'grid menu:grid-cols-4',
+  },
+  inline_width: { child: 'menu:w-max', parent: '' },
+};
+
+/*
+ * Do not delete this code.
+ * Utilized by Tailwind to load dynamic classes later.
+ */
+const possible = [
```
Comment (mcortes19) [inline src/lib/components/02-components/menus/main-menu/main-menu.tsx:122]: This variable is not used, could you please review it

**Review by mcortes19 — CHANGES_REQUESTED** [review]
Great job @robherba! I just found some minor issues:

- On mobile,  font-size should be `18px`
- **Talk to an expert** button text should be `16px`
- Arrows are breaking in a new line when text is long Screenshot 2024-02-12 at 16 52 10

**Comment by mcortes19** [comment]
Please review conflicts

**Comment by mcortes19** [comment]
@robherba please add nav tag with aria label for the entire menu, see: https://www.w3.org/WAI/tutorials/menus/structure/

**Review by josue2591 — APPROVED** [review]
Some minor accessibility adjustments, other than looks good!

**Pull request merged**